runtime/tchannel_client.go

```diff
@@ -97,59 +106,80 @@ func (c *tchannelClient) writeArgs(call *tchannel.OutboundCall, headers map[stri
 		return false, nil, errors.Wrapf(err, "could not create arg2reader for outbound call response: %s", c.serviceName)
 	}
 
-	headers, err := ReadHeaders(treader)
-	if err != nil {
+	if respHeaders, err = ReadHeaders(treader); err != nil {
 		_ = treader.Close()
-
 		return false, nil, errors.Wrapf(err, "could not read headers for outbound call response: %s", c.serviceName)
 	}
 
-	if err := EnsureEmpty(treader, "reading response headers"); err != nil {
+	if err = EnsureEmpty(treader, "reading response headers"); err != nil {
 		_ = treader.Close()
-
 		return false, nil, errors.Wrapf(err, "could not ensure arg2reader is empty for outbound call response: %s", c.serviceName)
 	}
 
-	if err := treader.Close(); err != nil {
+	if err = treader.Close(); err != nil {
 		return false, nil, errors.Wrapf(err, "could not close arg2reader for outbound call response: %s", c.serviceName)
 	}
 
-	success := !response.ApplicationError()
-	treader, err = response.Arg3Reader()
-	if err != nil {
-		return success, headers, errors.Wrapf(err, "could not create arg3Reader for outbound call response: %s", c.serviceName)
+	if treader, err = response.Arg3Reader(); err != nil {
+		return false, respHeaders, errors.Wrapf(err, "could not create arg3Reader for outbound call response: %s", c.serviceName)
 	}
 
-	if err := ReadStruct(treader, resp); err != nil {
+	if err = ReadStruct(treader, resp); err != nil {
 		_ = treader.Close()
-
-		return success, headers, errors.Wrapf(err, "could not read outbound call response: %s", c.serviceName)
+		return false, respHeaders, errors.Wrapf(err, "could not read outbound call response: %s", c.serviceName)
 	}
 
-	if err := EnsureEmpty(treader, "reading response body"); err != nil {
+	if err = EnsureEmpty(treader, "reading response body"); err != nil {
 		_ = treader.Close()
-
 		return false, nil, errors.Wrapf(err, "could not ensure arg3reader is empty for outbound call response: %s", c.serviceName)
 	}
 
-	return success, headers, treader.Close()
+	if err = treader.Close(); err != nil {
+		return false, nil, errors.Wrapf(err, "could not close arg3reader for outbound call response: %s", c.serviceName)
+	}
+
+	return !response.ApplicationError(), respHeaders, nil
 }
 
 // Call makes a RPC call to the given service.
-func (c *tchannelClient) Call(ctx context.Context, thriftService, methodName string, reqHeaders map[string]string, req, resp RWTStruct) (bool, map[string]string, error) {
-	var respHeaders map[string]string
-	var isOK bool
+func (c *tchannelClient) Call(ctx context.Context,
+	clientMethod, thriftService, thriftMethod string,
+	reqHeaders map[string]string,
+	req, resp RWTStruct,
+) (success bool, respHeaders map[string]string, err error) {
+	serviceMethod := fmt.Sprintf("%s::%s", thriftService, thriftMethod)
+	// TODO: Instead of creating method scope, add method call tags to subchannel stats reporter.
+	// Pending uber-go/tchannel PR to add custom tags to subchannel calls.
+	methodScope := c.scope.Tagged(map[string]string{
```

Comment:
This `methodScope` should be defined/subscoped in the outer layer, i.e, the client method definition, and accessed here, this way there is no need to pass around the `clientMethod` param.

Reply:
If I create the tagged 'methodScope' in the client method definition, then I'll need to pass the `methodScope` to the `Call()` method instead of the `clientMethod` string.

Reply:
Actually, on second thought, I'm going to collect the metrics in the generated client method, so we collect the complete client method latency, and no need to pass scope to TChannelClient.